npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -182,25 +182,36 @@ def inferred_npmrc # rubocop:disable Metrics/PerceivedComplexity
 
       sig { returns(T.nilable(T.any(Integer, String))) }
       def npm_version
-        @npm_version ||= T.let(package_manager.setup("npm"), T.nilable(T.any(Integer, String)))
+        @npm_version ||= T.let(package_manager_helper.setup("npm"), T.nilable(T.any(Integer, String)))
       end
 
       sig { returns(T.nilable(T.any(Integer, String))) }
       def yarn_version
-        @yarn_version ||= T.let(package_manager.setup("yarn"), T.nilable(T.any(Integer, String)))
+        @yarn_version ||= T.let(package_manager_helper.setup("yarn"), T.nilable(T.any(Integer, String)))
       end
 
       sig { returns(T.nilable(T.any(Integer, String))) }
       def pnpm_version
-        @pnpm_version ||= T.let(package_manager.setup("pnpm"), T.nilable(T.any(Integer, String)))
+        @pnpm_version ||= T.let(package_manager_helper.setup("pnpm"), T.nilable(T.any(Integer, String)))
       end
 
-      sig { returns(PackageManager) }
-      def package_manager
-        @package_manager ||= T.let(PackageManager.new(
-                                     parsed_package_json,
-                                     lockfiles: { npm: package_lock || shrinkwrap, yarn: yarn_lock, pnpm: pnpm_lock }
-                                   ), T.nilable(PackageManager))
+      sig { returns(PackageManagerHelper) }
+      def package_manager_helper
```

Comment:
We change the naming as `package_manager_helper` to reduce confusion. 

- This class was used as utility to install `package manager` for `file_fetcher` 
- We created  Package Manager for each `npm`, `yarn` and `pnpm` and the helper is going to also help to create instance for package manager.